### ISSUE TITLE
FastAPI cursor based pagination example

### ISSUE THREAD

**Issue by shellking4** [tiangolo/fastapi#4991]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

def get_all_with_pagination(*, dB:Session = Dépends(database.get_db), page_number: int=1, page_size: int = 70):
  return engine_service.get_with_pagination(
 dB, page_number, page_size
)
```

### Description

I would like to have an example of how to implement a cursor based pagination in FastAPI. Any help would very welcome. I managed to implement offset based pagination like the one shown in the docs but I read on the internet that offset based paginations are inefficient. Then I would like to see how to do it in FastAPI + Sqlalchemy + PostgreSQL. The frontend is using datatable.net, how to put everything together to have a full working pagination system. Thank you for your help in advance

### Operating System

Linux

### FastAPI Version

0.75.0

### Python Version

3.10.4

**Comment by acidjunk** [comment]
We use the offset based pagination: 

https://github.com/tiangolo/full-stack-fastapi-postgresql/blob/490c554e23343eec0736b06e59b2108fdd057fdc/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/app/crud/base.py#L29

Satisfied about performance etc..
(even extended it with filters/search)

**Comment by shellking4** [comment]
> We use the offset based pagination:
> 
> https://github.com/tiangolo/full-stack-fastapi-postgresql/blob/490c554e23343eec0736b06e59b2108fdd057fdc/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/app/crud/base.py#L29
> 
> Satisfied about performance etc.. (even extended it with filters/search)

Thank you very much sir. I think I'll go for the offset one for my current project. Thank you for your reply

**Comment by acidjunk** [comment]
U R Welcome: you might like this even more: https://github.com/acidjunk/fastapi-postgres-boilerplate/blob/master/server/crud/base.py

**Comment by shellking4** [comment]
> U R Welcome: you might like this even more: https://github.com/acidjunk/fastapi-postgres-boilerplate/blob/master/server/crud/base.py

Yeah very helpful sir it gives me more insight on the implementation. Thank you sir
